pkg/sql/distsqlrun/outbox.go

```diff
@@ -203,6 +203,18 @@ func (m *outbox) mainLoop(ctx context.Context) error {
 		m.statsCollectionEnabled = true
 		span.SetTag(streamIDTagKey, m.streamID)
 	}
+	defer func() {
+		tracing.FinishSpan(span)
+		if trace := getTraceData(ctx); trace != nil {
+			err := m.addRow(ctx, nil, &ProducerMetadata{TraceData: trace})
+			if err != nil {
+				log.Error(ctx, err)
+			}
+		}
+		if err := m.flush(ctx); err != nil {
```

Comment:
Are we sure this (and `addRow`) is ok to do in all the `return` cases in the function?